msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CoreMLCompilerTaskBase.cs

```diff
@@ -0,0 +1,239 @@
+
+		#endregion
+
+		#region Outputs
+
+		[Output]
+		public ITaskItem[] BundleResources { get; set; }
+
+		[Output]
+		public ITaskItem[] PartialAppManifests { get; set; }
+
+		#endregion
+
+		string DeveloperRootBinDir {
+			get { return Path.Combine (SdkDevPath, "usr", "bin"); }
+		}
+
+		string GetFullPathToTool ()
+		{
+			if (!string.IsNullOrEmpty (ToolPath))
+				return Path.Combine (ToolPath, ToolExe);
+
+			var path = Path.Combine (DeveloperRootBinDir, ToolExe);
+
+			return File.Exists (path) ? path : ToolExe;
+		}
+
+		static ProcessStartInfo GetProcessStartInfo (IDictionary<string, string> environment, string tool, string args)
+		{
+			var startInfo = new ProcessStartInfo (tool, args);
+
+			startInfo.WorkingDirectory = Environment.CurrentDirectory;
+
+			foreach (var variable in environment)
+				startInfo.EnvironmentVariables[variable.Key] = variable.Value;
+
+			startInfo.CreateNoWindow = true;
+
+			return startInfo;
+		}
+
+		int Compile (ITaskItem item, string outputDir, string log, string partialPlist)
+		{
+			var environment = new Dictionary<string, string> ();
+			var args = new ProcessArgumentBuilder ();
+
+			args.Add ("compile");
+			args.AddQuoted (item.ItemSpec);
+			args.AddQuoted (Path.GetFullPath (outputDir));
+			args.Add ("--output-partial-info-plist");
+			args.AddQuoted (partialPlist);
+
+			var startInfo = GetProcessStartInfo (environment, GetFullPathToTool (), args.ToString ());
+			var errors = new StringBuilder ();
+			int exitCode;
+
+			try {
+				Log.LogMessage (MessageImportance.Normal, "Tool {0} execution started with arguments: {1}", startInfo.FileName, startInfo.Arguments);
+
+				using (var stdout = File.CreateText (log)) {
+					using (var stderr = new StringWriter (errors)) {
+						var process = ProcessUtils.StartProcess (startInfo, stdout, stderr);
+
+						process.Wait ();
+
+						exitCode = process.Result;
+					}
+
+					Log.LogMessage (MessageImportance.Low, "Tool {0} execution finished (exit code = {1}).", startInfo.FileName, exitCode);
+				}
+			} catch (Exception ex) {
+				Log.LogError ("Error executing tool '{0}': {1}", startInfo.FileName, ex.Message);
+				File.Delete (log);
+				return -1;
+			}
+
+			if (exitCode != 0) {
+				// Note: ibtool or actool exited with an error. Dump everything we can to help the user
+				// diagnose the issue and then delete the manifest log file so that rebuilding tries
+				// again (in case of ibtool's infamous spurious errors).
+				if (errors.Length > 0)
+					Log.LogError (null, null, null, item.ItemSpec, 0, 0, 0, 0, "{0}", errors);
+
+				Log.LogError ("{0} exited with code {1}", ToolName, exitCode);
+
+				// Note: If the log file exists, log those warnings/errors as well...
+				if (File.Exists (log)) {
+					Log.LogError (null, null, null, item.ItemSpec, 0, 0, 0, 0, "{0}", File.ReadAllText (log));
+					File.Delete (log);
+				}
+			}
+
+			return exitCode;
+		}
+
+		static string GetPathWithoutExtension (string path)
```

Comment:
Can't you use [System.IO.Path.GetFileNameWithoutExtension](https://msdn.microsoft.com/en-us/library/system.io.path.getfilenamewithoutextension(v=vs.110).aspx)?

Reply:
No, because GetFileNameWithoutExtension just gets the file name, not the path.

Reply:
The implementation here is wrong then, because with `/foo.bar/zap` it'll return `/foo`, which is clearly wrong.

Reply:
Fixed